src/Commands/CloneCommand.php

```diff
@@ -0,0 +1,217 @@
+     *
+     * @param array $methods
+     *
+     * @return CloneCommand
+     */
+    protected function setMethods(array $methods = [])
+    {
+        $this->methods = $methods;
+        return $this;
+    }
+
+    /**
+     * Prepare operation and execute.
+     *
+     * @param $operation
+     *
+     * @return bool
+     */
+    protected function doOperation($operation = '')
+    {
+        $opMethod = $this->formatOpMethod($operation);
+        $site = $this->getSite($this->siteName);
+
+        $progress = $this->startProgressBar($operation);
+        $workflow = $this->triggerWorkflow($site, $opMethod);
+
+        while (!$workflow->isFinished()) {
+            $this->doWorkflow($workflow, $progress);
+        }
+        $progress->finish();
+        return true;
+    }
+
+    /**
+     * Configure and trigger the workflow operation.
+     *
+     * @param \Terminus\Models\Site $site
+     *
+     * @param string $opMethod
+     *
+     * @return \Terminus\Models\Workflow
+     * @throws \Terminus\Exceptions\TerminusException
+     */
+    protected function triggerWorkflow(Site $site, $opMethod = '')
+    {
+        return $site->environments
+            ->get($this->targetEnv)
+            ->$opMethod($this->originEnv);
+    }
+
+    /**
+     * Setup the progress bar.
+     *
+     * @param string $operation
+     *
+     * @return \Symfony\Component\Console\Helper\ProgressBar
+     */
+    protected function startProgressBar($operation = '')
+    {
+        ProgressBar::setFormatDefinition('minimal', self::PROGRESS_FORMAT);
+        $progress = $this->io()->createProgressBar(100);
+        $progress->setFormat('minimal');
+        $progress->setMessage("Cloning {$operation}...");
+        $progress->start();
+        return $progress;
+    }
+
+    /**
+     * Refresh Workflow data, advance progress, and hangout.
+     *
+     * @param \Terminus\Models\Workflow $workflow
+     * @param \Symfony\Component\Console\Helper\ProgressBar $progress
+     *
+     * @return bool
+     */
+    protected function doWorkflow(Workflow $workflow, ProgressBar $progress)
+    {
+        $workflow->fetch();
+        $progress->advance(1);
+        sleep(1);
+        return true;
+    }
+
+    private function setOperations(array $options = [])
+    {
+        $this->operations = $options;
+        return $this;
+    }
+
+    /**
+     * @return array Operations to be performed.
+     */
+    private function getOperations()
+    {
+        return $this->operations;
+    }
+
+    /**
+     * Check to see that origin and target are from the same site.
+     *
+     * @param string $origin
+     * @param string $target
+     *
+     * @return bool
+     * @throws \Exception
+     */
+    private function validateSites($origin, $target)
+    {
+        if (explode('.', $origin)[0] != explode('.', $target)[0]) {
+            throw new \Exception("Clone operations are only available within the same site.", 1);
```

Comment:
If we do not permit clone operations between sites, the second parameter should not be a site/env, but just the name of an env.